### PR TITLE
feat:  set upper limit of relayFee to 1 BNB and add timelock for large cross-chain transfers

### PR DESCRIPTION
**Pull request by cosinlink**
### Description

set upper limit of relayFee to 1 BNB and upper limit of cross-chain transfer amount to 1000 BNB

### Rationale

for cross-chain safety

### Changes

Notable changes:
* set upper limit of relayFee and ackRelayFee to 1 BNB
* set upper limit of cross-chain transfer amount to 1000 BNB
* modify tests

### CODE REVIEW

File: contracts/Config.sol [hunk 1]
```diff
@@ -14,15 +14,15 @@ abstract contract Config {
...
-    address public constant PROXY_ADMIN = address(0);
-    address public constant GOV_HUB = address(0);
-    address public constant CROSS_CHAIN = address(0);
-    address public constant TOKEN_HUB = address(0);
-    address public constant LIGHT_CLIENT = address(0);
-    address public constant RELAYER_HUB = address(0);
-    address public constant BUCKET_HUB = address(0);
-    address public constant OBJECT_HUB = address(0);
-    address public constant GROUP_HUB = address(0);
+    address public constant PROXY_ADMIN = 0xc95ff0608561b6bA084c78D14f09e9826190f968;
```
Comment (guagualvcha) [inline contracts/Config.sol:17]: no need to change.

File: contracts/middle-layer/TokenHub.sol [hunk 2]
```diff
@@ -17,7 +17,7 @@ contract TokenHub is Config, ReentrancyGuardUpgradeable, IMiddleLayer {
 
     uint256 public constant MAX_GAS_FOR_TRANSFER_BNB = 5000;
     uint256 public constant REWARD_UPPER_LIMIT = 1e18;
-
+    uint256 public constant MAX_TRANSFER_AMOUNT = 1000 ether;
```
Comment (guagualvcha) [inline contracts/middle-layer/TokenHub.sol:20]: Please make it governable.

File: contracts/middle-layer/TokenHub.sol [hunk 3]
```diff
@@ -62,6 +62,11 @@ contract TokenHub is Config, ReentrancyGuardUpgradeable, IMiddleLayer {
         _;
     }
 
+    modifier transferAmountCheck(uint256 _amount) {
```
Comment (guagualvcha) [inline contracts/middle-layer/TokenHub.sol:65]: I think we still allow transfer larger than 1000BNB, but just time lock it if it exceed this.

File: contracts/middle-layer/TokenHub.sol [hunk 4]
```diff
@@ -148,7 +153,10 @@ contract TokenHub is Config, ReentrancyGuardUpgradeable, IMiddleLayer {
      * @param recipient The destination address of the cross-chain transfer on GNFD.
      * @param amount The amount to transfer
      */
-    function transferOut(address recipient, uint256 amount) external payable returns (bool) {
+    function transferOut(
+        address recipient,
+        uint256 amount
+    ) external payable transferAmountCheck(amount) returns (bool) {
```
Comment (guagualvcha) [inline contracts/middle-layer/TokenHub.sol:159]: I think no limitation here.

File: contracts/Config.sol (hunk 1 above)
Comment (cosinlink) [reply contracts/Config.sol]: fixed

File: contracts/middle-layer/TokenHub.sol (hunk 2 above)
Comment (cosinlink) [reply contracts/middle-layer/TokenHub.sol]: fixed

File: contracts/middle-layer/TokenHub.sol (hunk 3 above)
Comment (cosinlink) [reply contracts/middle-layer/TokenHub.sol]: fixed

File: contracts/middle-layer/TokenHub.sol (hunk 4 above)
Comment (cosinlink) [reply contracts/middle-layer/TokenHub.sol]: fixed

File: .env.example [hunk 5]
```diff
@@ -8,3 +8,7 @@ RelayerPrivateKey=0x03ec592c156849665fa319fc7a11960dbd76f59ed06f664103b58668fb78
 
 # create a new api-key on https://bscscan.com/myapikey, replace `EXAMPLEKEY` with your api-key,
 BSCSCAN_APIKEY=EXAMPLEKEY
+
+RPC_URL=https://bsc-dataseed1.binance.org
+OPERATOR_PRIVATE_KEY=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
```
Comment (guagualvcha) [inline .env.example:13]: do not have plain private key here.

File: contracts/interface/ITokenHub.sol [hunk 6]
```diff
@@ -3,16 +3,9 @@
 pragma solidity ^0.8.0;
 
 interface ITokenHub {
-    function transferOut(
```
Comment (guagualvcha) [inline contracts/interface/ITokenHub.sol:6]: This should not be deleted.

File: .env.example (hunk 5 above)
Comment (cosinlink) [reply .env.example]: fixed

File: contracts/interface/ITokenHub.sol (hunk 6 above)
Comment (cosinlink) [reply contracts/interface/ITokenHub.sol:6]: fixed

**Review by guagualvcha — APPROVED** [review]

**Review by yutianwu — APPROVED** [review]

**Pull request merged**